src/api/dart/api/room.rs

```diff
@@ -0,0 +1,410 @@
+use derive_more::From;
+use flutter_rust_bridge::{frb, DartOpaque};
+use tracerr::Traced;
+
+use crate::{
+    api::{api::ApiMediaStreamSettings, Error as DartError},
+    media::MediaSourceKind,
+    platform::{self, utils::dart_future::IntoDartFuture},
+    room as core,
+};
+
+#[cfg(doc)]
+use crate::room::Room;
+
+/// External handle to a [`Room`].
+#[derive(Debug, From)]
+#[frb(opaque)]
+pub struct RoomHandle(pub(crate) core::RoomHandle);
+
+// Only used on single thread
+unsafe impl Send for RoomHandle {}
+unsafe impl Sync for RoomHandle {}
```

Comment:
@alexlapa @evdokimovs I don't like such approach. It's better than to wrap `core::RoomHandle` into `send_wrapper::SendWrapper` instead.